app/src/ui/branches/pull-request-list.tsx

```diff
@@ -54,6 +48,27 @@ interface IPullRequestListState {
   readonly selectedItem: IPullRequestListItem | null
 }
 
+function resolveSelectedItem(
+  group: IFilterListGroup<IPullRequestListItem>,
+  props: IPullRequestListProps,
+  currentlySelectedItem: IPullRequestListItem | null
+): IPullRequestListItem | null {
+  let selectedItem: IPullRequestListItem | null = null
+
+  if (props.selectedPullRequest != null) {
```

Comment:
The important thing about this change is that we prioritize `props` changes over the component state. `componentWillReceiveProps` didn't do this - it favoured `this.state.selectedItem` which meant it could ignore changes from the parent component while re-rendering.

Also, extracting this out and using it in both places for some consistency...